Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -486,127 +471,116 @@ static inline Marker read_marker_at_cursor(InputMemoryStream& stream)
-        return false;
+        return Error::from_string_literal("Unsupported number of components");
     }
 
     for (int i = 0; i < component_count; i++) {
         u8 component_id = 0;
         stream >> component_id;
-        if (stream.handle_any_error())
-            return false;
+        TRY(stream.try_handle_any_error());
 
         auto& component = context.components[i];
         if (component.id != component_id) {
             dbgln("JPEG decode failed (component.id != component_id)");
-            return false;
+            return Error::from_string_literal("JPEG decode failed (component.id != component_id)");
         }
 
         u8 table_ids = 0;
         stream >> table_ids;
-        if (stream.handle_any_error())
-            return false;
+        TRY(stream.try_handle_any_error());
 
         component.dc_destination_id = table_ids >> 4;
         component.ac_destination_id = table_ids & 0x0F;
 
         if (context.dc_tables.size() != context.ac_tables.size()) {
             dbgln_if(JPG_DEBUG, "{}: DC & AC table count mismatch!", stream.offset());
-            return false;
+            return Error::from_string_literal("DC & AC table count mismatch");
         }
 
         if (!context.dc_tables.contains(component.dc_destination_id)) {
             dbgln_if(JPG_DEBUG, "DC table (id: {}) does not exist!", component.dc_destination_id);
-            return false;
+            return Error::from_string_literal("DC table does not exist");
         }
 
         if (!context.ac_tables.contains(component.ac_destination_id)) {
             dbgln_if(JPG_DEBUG, "AC table (id: {}) does not exist!", component.ac_destination_id);
-            return false;
+            return Error::from_string_literal("AC table does not exist");
         }
     }
 
     u8 spectral_selection_start = 0;
     stream >> spectral_selection_start;
-    if (stream.handle_any_error())
-        return false;
+    TRY(stream.try_handle_any_error());
     u8 spectral_selection_end = 0;
     stream >> spectral_selection_end;
-    if (stream.handle_any_error())
-        return false;
+    TRY(stream.try_handle_any_error());
     u8 successive_approximation = 0;
     stream >> successive_approximation;
-    if (stream.handle_any_error())
-        return false;
+    TRY(stream.try_handle_any_error());
     // The three values should be fixed for baseline JPEGs utilizing sequential DCT.
     if (spectral_selection_start != 0 || spectral_selection_end != 63 || successive_approximation != 0) {
         dbgln_if(JPG_DEBUG, "{}: ERROR! Start of Selection: {}, End of Selection: {}, Successive Approximation: {}!",
             stream.offset(),
             spectral_selection_start,
             spectral_selection_end,
             successive_approximation);
-        return false;
+        return Error::from_string_literal("spectral_selection_start != 0 || spectral_selection_end != 63 || successive_approximation != 0");
```

Comment:
Same note here about using an actual error message and not just copying the conditional